### PR TITLE
Add jira venue using pandoc new jira output format

### PR DESCRIPTION
**Pull request by cderv**
This is a PR to offer an alternative to current proposal in #265 to deal with #193 , (and it is inspired by it)

Instead of adding a custom post processing from md file produce by `reprex_render` to transform to JIRA markup, it is build on new jira output format added in pandoc 2.7.3. This allows to have a full conversion of markdown syntax to JIRA markup. This seems more robust to build on pandoc, with the drawback of requiring last pandoc version 2.7.3 (not included in RStudio IDE yet) to be able to work. 

I added a new shortcut `reprex_jira` too and some test, showing that pandoc is able to convert markdown syntax to jira markup. 

Also, it is rebased already on new dev reprex codebase.

One question: you don't seem to use `@importFrom`, so I did not add the two `rmarkdown` function into NAMESPACE.

### CODE REVIEW

File: R/reprex_impl.R [hunk 1]
```diff
@@ -109,6 +109,20 @@ reprex_impl <- function(x_expr = NULL,
     reprex_file <- rtf_file
   }
 
+  if (venue == "jira") {
+    jira_file <-  files[["jira_file"]]
+    # jira support for pandoc conversion was added in version 2.7.3
+    if (!rmarkdown::pandoc_available("2.7.3")) {
```
Comment (jennybc) [inline R/reprex_impl.R]: Let's handle this more like `venue = "rtf"` and highlight. That is:

  * Put the test in a helper `jira_requires_pandoc_2_7_3()`.
  * Check this much earlier.

https://github.com/tidyverse/reprex/blob/b5a049e1660652468715a36d169b379aa1edf29e/R/highlight.R#L16-L25

https://github.com/tidyverse/reprex/blob/b5a049e1660652468715a36d169b379aa1edf29e/R/reprex_impl.R#L20

Comment (cderv) [reply R/reprex_impl.R]: I missed that. This is handle now. I added the help in utils instead of a `jira.R` file with only one function.

**Comment by jennybc** [comment]
@cderv are you still invested in this? In any case, there was massive change in reprex in the meantime, so I'm going to close this PR.

**Comment by cderv** [comment]
I will not more use directly as I would before but I still think it would be useful to some. (like for arrow that uses JIRA)

Pandoc supporting Jira/Confluence wiki markup in recent version, I think it should not be so difficult but I know less the internal now. Do you want me to try on another PR ?

**Comment by jennybc** [comment]
If reprex doesn't need to add much complexity, it feels like Jira **is** relevant enough to consider as a `venue`. We have a bit of a chicken-egg problem re: knowing how hard it is vs. who would use it. But based on this PR, it doesn't seem like it would be too bad.

**Comment by jennybc** [comment]
I've already decided what else is "in" for v1.0.0, which is basically just the Slack `venue`. But I plan to make a v1.1.0 release shortly thereafter, which could include Jira. So we don't have to decide right this minute.

**Pull request closed without merging**